register/views.py

```diff
@@ -1,31 +1,75 @@
-from django.http import HttpResponse
-from django.shortcuts import render
-from .forms import StudentRegisterForm, AdminStaffRegisterForm
-from OneApply.constants import UserType
-
-
-def register_user(request, user_type):
-    form = None
-    if request.method == "POST":
-        if user_type == UserType.STUDENT:
-            form = StudentRegisterForm(request.POST)
-        elif user_type == UserType.ADMIN_STAFF:
-            form = AdminStaffRegisterForm(request.POST)
-        if form is not None and form.is_valid():
-            password = form.cleaned_data["input_password"]
-            f = form.save(commit=False)
-            f.password = password
-            f.save()
-            return HttpResponse("Registered Successfully!")
-    else:
-        if user_type == UserType.STUDENT:
-            form = StudentRegisterForm()
-        elif user_type == UserType.ADMIN_STAFF:
-            form = AdminStaffRegisterForm()
-    context = {
-        "form": form,
-        "user_type": user_type,
-        "constant_ut_student": UserType.STUDENT,
-        "constant_ut_adminStaff": UserType.ADMIN_STAFF,
-    }
-    return render(request, "register/index.html", context)
+from django.http import HttpResponse
+from django.shortcuts import render
+from django.core.mail import EmailMessage
+from django.contrib.sites.shortcuts import get_current_site
+from django.utils.encoding import force_bytes, force_text
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from django.template.loader import render_to_string
+from register.models import Student
+from .forms import StudentRegisterForm, AdminStaffRegisterForm
+from .tokens import account_activation_token
+from OneApply.constants import UserType
+
+
+def register_user(request, user_type):
+    form = None
+    if request.method == "POST":
+        if user_type == UserType.STUDENT:
```

Comment:
can u use switch-case for the user_type?

Reply:
can you explain a bit more as to why switch-case would be better?

Reply:
switch cases are considered better instead of nested-if. But since we just have 2 case conditions for now, you can keep it as it is.